core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -186,4 +224,43 @@ ManifestFile manifest() {
       return manifest;
     }
   }
+
+  static class ContentFileStructWithMetrics implements StructLike {
+    private final int structSize;
+    private final StructLike fileAsStruct;
+    private final MetricsUtil.ReadableMetricsStruct readableMetrics;
+
+    ContentFileStructWithMetrics(
+        int structSize,
```

Comment:
Do we need to pass this? StructLike has a size() operator

Reply:
I tried to change the code and didnt work.  This is actually the expectedSize (to use the terminology in other places, aka the user-requested size).  The fileAsStruct.size() is actually always all the possible DataFile values, if they are not projected they are just null..